module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1394,18 +1397,38 @@ public function renewMyItems($renewDetails)
      */
     public function getPickupLocations($patron, $holdInfo = null)
     {
+        $limitedServicePoints = null;
+        if (
+            str_contains($this->config['Holds']['limitPickupLocations'] ?? '', 'itemEffectiveLocation')
+            // If there's no item ID, it must be a title-level hold,
+            // so limiting by itemEffectiveLocation does not apply
+            && $holdInfo['item_id'] ?? false
+        ) {
+            $response = $this->makeRequest(
```

Comment:
I think it would make sense to create a `protected getItemById()` method to abstract this lookup, since we're now doing it in two different places.